tgui/packages/tgui/interfaces/PowerMonitor/Smes.tsx

```diff
@@ -55,31 +57,117 @@ export const PowerMonitorSmesGlobal = (_props, context) => {
   return (
     <>
-      <Table.Cell header>Area</Table.Cell>
-      <Table.Cell header>Stored Power</Table.Cell>
-      <Table.Cell header>Charging</Table.Cell>
-      <Table.Cell header>Input</Table.Cell>
-      <Table.Cell header>Output</Table.Cell>
-      <Table.Cell header>Active</Table.Cell>
-      <Table.Cell header>Load</Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.Area ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.Area)} >
+          Area
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.StoredPower ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.StoredPower)}>
+          Stored Power#
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.Charging ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.Charging)} >
+          Charging
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.Input ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.Input)}>
+          Input
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.Output ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.Output)}>
+          Output
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.Active ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.Active)}>
+          Active
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === SmesTableHeaderColumns.Load ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(SmesTableHeaderColumns.Load)} >
+          Load
+        </Header>
+      </Table.Cell>
     </>
   );
 };
 
+const smesDataComparer
+  = (a: PowerMonitorSmesItemData,
+    b: PowerMonitorSmesItemData,
+    names: Record<string, string>,
+    field: SmesTableHeaderColumns): number => {
+
+    if (field === SmesTableHeaderColumns.Area) {
+      return names[a[field]].localeCompare(names[b[field]]);
+    } else if (field === SmesTableHeaderColumns.Charging || field === SmesTableHeaderColumns.Active) {
+      if (a[field] === a[field]) {
+        return 0;
+      } else if (a[field] > b[field]) {
+        return 1;
+      } else {
+        return -1;
+      }
```

Comment:
If this isn't a string (should be using `localeCompare` as above), or a number (should be using `numericCompare`, as below), what are the types of these fields? Based on context assuming booleans. In either case, extracting out and naming the type of comparison with a comparator function is still a good idea.